app/lib/dfe/bigquery/application_metrics_by_provider.rb

```diff
@@ -0,0 +1,115 @@
+
+                          number_of_candidates_with_reconfirmed_offers_deferred_from_previous_cycle_to_date
+                          number_of_candidates_with_reconfirmed_offers_deferred_from_previous_cycle_to_same_date_previous_cycle
+                          number_of_candidates_with_reconfirmed_offers_deferred_from_previous_cycle_to_date_as_proportion_of_last_cycle
+
+                          number_of_candidates_who_had_all_applications_rejected_this_cycle_to_date
+                          number_of_candidates_who_had_all_applications_rejected_this_cycle_to_same_date_previous_cycle
+                          number_of_candidates_who_had_all_applications_rejected_this_cycle_to_date_as_proportion_of_last_cycle
+
+                          number_of_candidates_who_had_an_inactive_application_this_cycle_to_date
+                          number_of_candidates_who_had_an_inactive_application_this_cycle_to_date_as_proportion_of_submitted_candidates].freeze
+
+      attr_reader :cycle_week,
+                  :recruitment_cycle_year,
+                  :nonprovider_filter,
+                  :provider_filter,
+                  :provider_filter_category
+
+      def initialize(cycle_week:, provider_id: nil, recruitment_cycle_year: RecruitmentCycle.current_year)
+        @provider_id = provider_id&.to_s
+        @cycle_week = cycle_week
+        @recruitment_cycle_year = recruitment_cycle_year
+      end
+
+      def table_name
+        :'dataform.application_metrics_by_provider'
+      end
+
+      ### Candidate All query
+
+      def provider_data
+        query(provider_data_query)
+      end
+
+      def provider_data_query
+        select(SELECT_COLUMNS.join(', '))
+        .where(
+          'provider.id': @provider_id,
+          cycle_week:,
+          recruitment_cycle_year:,
+        ).where(<<~SQL.chomp).to_sql
+          (
+            nonprovider_filter_category = "Secondary subject"
+            OR (nonprovider_filter_category = "Level" AND nonprovider_filter IN ("Primary", "Secondary"))
+            OR nonprovider_filter = "All"
+          )
+        SQL
+      end
+
+      ### National queries
+
+      def national_data
+        query(national_data_query)
+      end
+
+      def national_data_query
+        select(SELECT_COLUMNS.join(', '))
+        .where(
+          cycle_week:,
+          recruitment_cycle_year:,
+          teach_first_or_iot_filter: 'All',
+          provider_filter_category: 'All',
+        ).where(<<~SQL.chomp).to_sql
+          (
+            nonprovider_filter_category = "Secondary subject"
+            OR (nonprovider_filter_category = "Level" AND nonprovider_filter IN ("Primary", "Secondary"))
+            OR (nonprovider_filter = "All")
+          )
+        SQL
+      end
+
+    private
+
+      def result_class = self.class::Result
+
+      class Result
+        attr_reader(*SELECT_COLUMNS.dup.delete_if { |s| s[/\./] }.push('provider_id'))
```

Comment:
👍 
I know we talked about this one, it's a bit awkward, but I think this is the most straightforward way to handle it. i think when we look at this next year, we'll know what was going on here and that's what's most important